web/src/api/stores/file.store.ts

```diff
@@ -41,34 +44,43 @@ export const useFileStore = defineStore("files", () => {
       file.astAndMappingLoaded = true;
       file.amountOfKgrams = file.amountOfKgrams || file.ast.length;
 
-      if (anonymize) {
-        const split = row.path!.split(".");
-        const extension = split[split.length - 1];
-        const name = randomNameGenerator();
-        row.path = `${name}/exercise.${extension}`;
-        extra.fullName = name;
-
-        const label = labelMap.get(extra.labels) || labelCounter;
-        if (!labelMap.has(extra.labels)) {
-          labelCounter += 1;
-        }
-        labelMap.set(extra.labels, label);
-        extra.labels = label;
-
-        if (extra.timestamp) {
-          extra.timestamp = new Date(extra.timestamp.getTime() + timeOffset);
-        }
+      // Add the label to the labels list, if it doesn't exist yet.
+      if (!labels.includes(row.label)) {
+        labels.push(extra.labels);
       }
 
+      // Store pseudo details.
+      const pseudoName = randomNameGenerator();
+      file.pseudo = {
+        path: `/exercise/${pseudoName}.${filePathExtension}`,

```

Comment:
Is the (full) path shown in the UI?

If so, showing an anonymized version is not really useful. I would prefer a message instead, e.g. "Full path not available when anonymization is enabled."